scheduler/pkg/klcpermit/workflow_manager_test.go

```diff
@@ -73,3 +76,258 @@ func Test_getCRDName(t *testing.T) {
+			want1:               false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, got1 := getLabelOrAnnotation(tt.pod, tt.primaryAnnotation, tt.secondaryAnnotation)
+			if got != tt.want {
+				t.Errorf("getLabelOrAnnotation() got = %v, want %v", got, tt.want)
+			}
+			if got1 != tt.want1 {
+				t.Errorf("getLabelOrAnnotation() got1 = %v, want %v", got1, tt.want1)
+			}
+		})
+	}
+}
+
+func Test_getSpan_unbindSpan(t *testing.T) {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				WorkloadAnnotation: "myworkload",
+				VersionAnnotation:  "0.0.1",
+				AppAnnotation:      "myapp",
+			},
+		},
+	}
+
+	r := NewWorkloadManager(nil)
+
+	_, span := r.getSpan(context.TODO(), &unstructured.Unstructured{}, pod)
+
+	require.NotNil(t, span)
+	require.Len(t, r.bindCRDSpan, 1)
+
+	_, span2 := r.getSpan(context.TODO(), &unstructured.Unstructured{}, pod)
+
+	require.Equal(t, span, span2)
+	require.Len(t, r.bindCRDSpan, 1)
+
+	pod2 := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				WorkloadAnnotation: "myworkload",
+				VersionAnnotation:  "0.0.2",
+				AppAnnotation:      "myapp",
+			},
+		},
+	}
+
+	_, span3 := r.getSpan(context.TODO(), &unstructured.Unstructured{}, pod2)
+
+	require.NotNil(t, span3)
+	require.Len(t, r.bindCRDSpan, 2)
+
+	_, span4 := r.getSpan(context.TODO(), &unstructured.Unstructured{}, pod2)
+
+	require.Equal(t, span3, span4)
+	require.Len(t, r.bindCRDSpan, 2)
+
+	_, span5 := r.getSpan(context.TODO(), &unstructured.Unstructured{}, pod)
+
+	require.Equal(t, span, span5)
+	require.Len(t, r.bindCRDSpan, 2)
+
+	r.unbindSpan(pod)
+	require.Len(t, r.bindCRDSpan, 1)
+
+	_, span6 := r.getSpan(context.TODO(), &unstructured.Unstructured{}, pod2)
+
+	require.Equal(t, span3, span6)
+	require.Len(t, r.bindCRDSpan, 1)
+
+	_, span7 := r.getSpan(context.TODO(), &unstructured.Unstructured{}, pod)
+
+	require.Equal(t, span, span7)
+	require.Len(t, r.bindCRDSpan, 2)
+
+}
+
+func Test_calculateVersion(t *testing.T) {
+	tests := []struct {
+		name string
+		pod  *corev1.Pod
+		want string
+	}{
+		{
+			name: "empty pod",
+			pod:  &corev1.Pod{},
+			want: "2166136261",
+		},
+		{
+			name: "no containers",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pod-name",
+				},
+			},
+			want: "2166136261",
+		},
+		{
+			name: "single container",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pod-name",
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "container-name",
+							Image: "image:tag",
+						},
+					},
+				},
+			},
+			want: "tag",
+		},
+		{
+			name: "single container latest tag",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pod-name",
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "container-name",
+							Image: "image:latest",
+						},
```

Comment:
the function looks also at the `Env` field, please add an entry to cover that behavior as well

Reply:
added, good catch